core/v1alpha1/extension_conversion.go

```diff
@@ -0,0 +1,63 @@
+// Copyright Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
+	v1 "maistra.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+// ConvertTo converts this SME to the Hub version (v1)
+func (src *ServiceMeshExtension) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1.ServiceMeshExtension)
+	return Convert_v1alpha1_ServiceMeshExtension_To_v1_ServiceMeshExtension(src, dst, nil)
+}
+
+// ConvertFrom converts from the Hub version (v1) to this version.
+func (dst *ServiceMeshExtension) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*v1.ServiceMeshExtension)
+	return Convert_v1_ServiceMeshExtension_To_v1alpha1_ServiceMeshExtension(src, dst, nil)
+}
+
+func Convert_v1_ServiceMeshExtensionSpec_To_v1alpha1_ServiceMeshExtensionSpec(src *v1.ServiceMeshExtensionSpec, dst *ServiceMeshExtensionSpec, scope apiconversion.Scope) error {
+	res, err := src.Config.MarshalJSON()
+	if err != nil {
+		return fmt.Errorf("config field could not be converted to string: %v", err)
+	}
+	dst.Config = string(res)
+
+	dst.Image = src.Image
+	dst.ImagePullPolicy = src.ImagePullPolicy
+	dst.ImagePullSecrets = src.ImagePullSecrets
+	dst.Phase = (*FilterPhase)(src.Phase)
+	dst.Priority = src.Priority
+	return Convert_v1_WorkloadSelector_To_v1alpha1_WorkloadSelector(&src.WorkloadSelector, &dst.WorkloadSelector, nil)
+}
+
+func Convert_v1alpha1_ServiceMeshExtensionSpec_To_v1_ServiceMeshExtensionSpec(src *ServiceMeshExtensionSpec, dst *v1.ServiceMeshExtensionSpec, scope apiconversion.Scope) error {
+	if err := dst.Config.UnmarshalJSON([]byte(src.Config)); err != nil {
```

Comment:
We should do something here to mitigate errors, as this could break the api server (this is the issue we had with 2.0.0 :().  I'm not sure if you can get away with doing something like:
```
  dst.Config = v1.ServiceMeshExtensionConfig{Data: map[string]interface{}{"text_config": src.Config}}
```
Then handle `text_config` as a special case when converting back to v1alpha1.

Reply:
done. Created https://issues.redhat.com/browse/MAISTRA-2391 to make use of this new field in the push logic.